src/flint/types/fmpz_mod.pyx

```diff
@@ -0,0 +1,357 @@
+            return NotImplemented
+
+        cdef fmpz_mod res
+        res = fmpz_mod.__new__(fmpz_mod)
+        res.ctx = self.ctx
+        fmpz_mod_add(
+            res.val, self.val, (<fmpz_mod>other).val, self.ctx.val
+        )
+        return res
+
+    def __radd__(self, other):
+        return self.__add__(other)
+
+    def __sub__(self, other):
+        other = self.any_as_fmpz_mod(other)
+        if other is NotImplemented:
+            return NotImplemented
+
+        cdef fmpz_mod res
+        res = fmpz_mod.__new__(fmpz_mod)
+        res.ctx = self.ctx
+
+        fmpz_mod_sub(
+            res.val, self.val, (<fmpz_mod>other).val, self.ctx.val
+        )
+        return res
+
+    def __rsub__(self, other):
+        return self.__sub__(other).__neg__()
+
+    def __mul__(self, other):
+        other = self.any_as_fmpz_mod(other)
+        if other is NotImplemented:
+            return NotImplemented
+
+        cdef fmpz_mod res
+        res = fmpz_mod.__new__(fmpz_mod)
+        res.ctx = self.ctx
+
+        fmpz_mod_mul(
+            res.val, self.val, (<fmpz_mod>other).val, self.ctx.val
+        )
+        return res
+
+    def __rmul__(self, other):
+        return self.__mul__(other)
+
+    @staticmethod
+    def _div_(left, right):
+        cdef bint check
+        cdef fmpz_mod res
+        res = fmpz_mod.__new__(fmpz_mod)
+        
+        # Division when left and right are fmpz_mod
+        if typecheck(left, fmpz_mod) and typecheck(right, fmpz_mod):
+            res.ctx = (<fmpz_mod>left).ctx
+            if not (<fmpz_mod>left).ctx == (<fmpz_mod>right).ctx:
+                raise ValueError("moduli must match")
+            check = fmpz_mod_divides(
+                res.val, (<fmpz_mod>left).val, (<fmpz_mod>right).val, res.ctx.val
+            ) 
+        
+        # Case when only left is fmpz_mod
+        elif typecheck(left, fmpz_mod):
+            res.ctx = (<fmpz_mod>left).ctx
+            right = any_as_fmpz(right)
+            if right is NotImplemented:
+                return NotImplemented   
+            check = fmpz_mod_divides(
+                res.val, (<fmpz_mod>left).val, (<fmpz>right).val, res.ctx.val
+            ) 
+
+        # Case when right is an fmpz_mod
+        else:
+            res.ctx = (<fmpz_mod>right).ctx
+            left = any_as_fmpz(left)
+            if left is NotImplemented:
+                return NotImplemented   
+            check = fmpz_mod_divides(
+                res.val, (<fmpz>left).val, (<fmpz_mod>right).val, res.ctx.val
+            ) 
+        
+        if check == 0:
+            raise ZeroDivisionError(f"{right} is not invertible modulo {res.ctx.modulus()}")
+
+        return res
+
+    def __truediv__(s, t):
+        return fmpz_mod._div_(s, t)
+
+    def __rtruediv__(s, t):
+        return fmpz_mod._div_(t, s)
+
+    def __floordiv__(self, other):
+        return NotImplemented
+
+    def inverse(self, check=True):
+        """
+        Computes a^-1 mod N
```

Comment:
It should be possible to use LaTeX here but possibly the LaTeX Sphinx extension is needed but not enabled...

Reply:
Yeah I'm trying to do that but the naive 
```
math:`a^{-1} \mod N`
```
isn't working at least...

Reply:
Oh, I got it.